### PR TITLE
Some PyPI packaging updates and other cleanup

### PR DESCRIPTION
**Pull request by rickecon**
@jdebacker This PR is the result of me going carefully through the [Packaging Python Projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/) tutorial, which led me to find some other cleanup items. This PR does the following.
* Changed the format of `setup.py` to be more consistent with the [Packaging Python Projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/) tutorial. The tutorial has a warning to not use `distutils.core` as it will be removed from the standard library in Python 3.12. I also made some other changes to the format of the `setup.py`.
* Added `pyproject.toml` exactly as detailed in the [Packaging Python Projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/) tutorial. I did this in the hope that some of the weird package upload problems we experienced with the Google Colab notebooks would go away (e.g., `mkl` not being present, even after pip install).
* Added the new `/tests/` folder location to be ignored in the `codecov.yml`.
* Remove some pre-alpha qualifying language from the `README.md` and added a `pip install` use case in the "Using/contributing to OG-Core" section.
* Get's rid of some erroneous OG-USA references (`PSL_catalogue.json`, `intro/intro.md`, `government.md`). I noticed the need for this when looking at the [Libraries.io](https://libraries.io/pypi/ogcore) statistics page linked to from the OG-Core PyPI page](https://pypi.org/project/ogcore/). The [libraries.io/pypi/ogcore](https://libraries.io/pypi/ogcore) statistics page on our package links to an old PyPI ogusa package page that has old BYU/OSPC documentation README.md. Everything else is correct, though. That made me think that there was a weird `ogusa` reference in the OG-Core repo.
* Version patch increases the version to 0.8.1 (`Makefile`, `__init__.py`, `setup.py`). I think the version patch update is important here if these changes update the package integrity when installing via `pip install ogcore`.

### CODE REVIEW

**Comment by jdebacker** [comment]
@rickecon If you have tested all these changes to `setup.py` and the package builds fine, then I just have the one small comment I left.  Please confirm things are tested, respond to my comment, then let me know you are ready to merge.  Thanks!

**Comment by rickecon** [comment]
@jdebacker. I can't see you comment anywhere. As soon as I find that, I can respond. Also, the following comments and questions.
1. The package successfully builds on my machine when I type `python3 -m build`. I get the following message, and the two package files referenced are successfully located in the `./dist/` folder.
```
Successfully built ogcore-0.8.1.tar.gz and ogcore-0.8.1-py3-none-any.whl
```
2. `test.pypi.org` is not allowing me to upload the package for testing when I type `python3 -m twine upload --repository testpypi dist/*` because of the following error, caused by package name overlap.
```
HTTPError: 403 Forbidden from https://test.pypi.org/legacy/
The user 'rickecon' isn't allowed to upload to project 'ogcore'. See
https://test.pypi.org/help/#project-name for more information.
```
3. How do I test the created packages if not on test.pypi.org?
4. How do I update the changelog and version on GitHub after this PR goes through?

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ clean:
 
 .PHONY=package
```
[the remark this answers is not in the document]
Comment (jdebacker) [reply Makefile:29]: @rickecon This is only comment - hadn't submitted review before.

**Comment by rickecon** [comment]
@jdebacker. Per your comment, I deleted the line 29-31 from `Makefile`. I still have questions 3-5 above.

**Comment by jdebacker** [comment]
@rickecon writes:

> The package successfully builds on my machine when I type `python3 -m build`

This is probably fine, but the GH Action in this repo uses `python setup.py sdist bdist_wheel`, so you might want to try that.

> How do I test the created packages if not on `test.pypi.org`?

If you want to use `test.pypi.org` you need to have an account there (separate from the PyPI.org account).

> How do I update the changelog and version on GitHub after this PR goes through?

To publish a new release on GH, follow [these instructions](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository).  OG-Core doesn't have a changelog.  It should.  [Here's an example](https://github.com/PSLmodels/Tax-Calculator/blob/master/docs/about/releases.md#2021-07-17-release-320) one from Tax-Calculator (this is manually updated with each release).  Right now, we just publish release notes with each (or at least most) releases.

> Why do we use the dynamic metadata setup.py approach which is "possibly deterministic" and "should be used only as an escape hatch when absolutely necessary"? "setup.py used to be required, but can be omitted with newer versions of setuptools and pip. The static metadata approach of a setup.cfg file is "guaranteed to be the same every time. This is simpler, easier to read, and avoids many common errors, like encoding errors." (all quotes from the [Packaging Python Projects](https://packaging.python.org/en/latest/tutorials/packaging-projects/) tutorial)

I don't know.  We've started with what was done in the Tax-Calculator repos.  Some of that might be necessary for conda packages, but I don't know.  I would say that as long as it works with PyPI, we should have a format that also allows us to create a conda package in case we'd like to (many developed projects publish to both).

**Comment by rickecon** [comment]
@jdebacker. I still was not able to upload my built packages to test.pypi.org due to a name conflict. But I was able to create both the `.whl` and `.tar.gz` package information, and I `pip install ./dist/ogcore-0.8.1.tar.gz`-ed it and ran the `run_ogcore_example.py` file and it ran great. This PR is ready for final review and merge.

**Pull request merged**